integration-tests/sync-tests/2-sync-replica.spec.ts

```diff
@@ -0,0 +1,121 @@
+import chai, { expect } from 'chai'
+import { Wallet, Contract, ContractFactory, providers } from 'ethers'
+import { ethers } from 'hardhat'
+import { injectL2Context } from '@eth-optimism/core-utils'
+
+import { sleep, l2Provider, replicaProvider } from '../test/shared/utils'
+import { OptimismEnv } from '../test/shared/env'
+import { DockerComposeNetwork } from '../test/shared/docker-compose'
+
+describe('Syncing a replica', () => {
+  let env: OptimismEnv
+  let wallet: Wallet
+  let replica: DockerComposeNetwork
+  let provider: providers.JsonRpcProvider
+
+  const sequencerProvider = injectL2Context(l2Provider)
+
+  /* Helper functions */
+
+  const startReplica = async () => {
```

Comment:
This function is basically the same as `startVerifier` - it should be possible to have a common file where both the verifier and replica can use that deduplicates the code

Reply:
^agreed, will refactor this!

Reply:
@tynes i switched both to poll the endpoint instead of waiting for docker logs! as a long term vision when both verifier and replicas are stable, i think these tests can go for a larger refactor in which they share the same test suite of transactions and we just use a `verifier` or `replica` switch to run each. 

as of now though, the replica has matching state roots for pretty much everything but the verifier is the opposite, so i'm keeping them as separate files (also not in the `test` folder for now) so we could buff up the replica tests before uniswap launch.

Reply:
Ok! Thats a good idea!